text/0000-default-field-values.md

```diff
@@ -0,0 +1,1802 @@
+  `match`es, `let` statements, and `panic!(..)`s will be allowed.
+  Another feasible extension in the future is allocation.
+
+  Therefore, constant expressions should be enough to satisfy most expressive
+  needs.
+
+## Instead of `Foo { ..Default::default() }`
+
+As an alternative to the proposed design is either explicitly writing out
+`..Default::default()` or extending the language such that `Foo { .. }` becomes
+sugar for `Foo { ..Default::default() }`. While the latter idea does not satisfy
+any of the [motivation] set out, the former does to a small extent.
+
+In particular, `Foo { .. }` as sugar slightly improves ergonomics.
+However, it has some notable problems:
+
++ Because it desugars to `Foo { ..Default::default() }`, it cannot be required
+  that the expression is a constant one. This carries all the problems noted in
+  the previous section on why default field values should be a `const` context.
+
++ There is no way of implementing a `Default` implementation that has mandatory
+  fields for users to specify during value construction.
+
++ It provides zero improvements to the ergonomics of *specifying* defaults,
+  only for using them. Arguably, the most important aspect of this RFC is
+  not the syntax `Foo { .. }` but rather the ability to provide default values
+  for field.
+
++ By extension, the improvement to documentation clarity is lost.
+
++ The trait `Default` must now become a `#[lang_item]`. This is a sign of
+  increasing the overall magic in the system; meanwhile, this proposal makes
+  the default values provided usable by other custom derive macros.
+
+Thus in conclusion, while desugaring `..` to `Default::default()` has lower cost,
+it also provides significantly less value to the point of not being worth it.
+
+## `..` is useful as a marker
+
+One possible change to the current design is to permit filling in defaults
+by simply writing `Foo {}`; in other words, `..` is simply dropped from the
+expression.
+
+Among the benefits are:
+
++ To enhance ergonomics of initialization further.
+
++ To introduce less syntax.
+
++ To be more in line with how other languages treat default values.
+
+Among the drawbacks are:
+
++ The syntax `Foo { .. }` is no longer introduced to complement the identical
+  pattern syntax. As aforementioned, destruction (and pattern matching)
+  generally attempts to follow construction in Rust. Because of that,
+  introducing `Foo { .. }` is essentially cost-free in terms of the complexity
+  budget. It is arguably even cost-negative.
+
++ By writing `Foo { .. }`, you have there is explicit indication that default
+  values are being used; this enhances local reasoning further.
+
+This RFC requires the `..` to get defaulted fields because it wants to continue
+to allow the workflow of intentionally *not* including `..` in the struct
+literal expression so that when a user adds a field they get compilation errors
```

Comment:
What about structs that are forever stable and the default field is something benign like PhantomData? My go-to example is euclid's types, where the extra overhead of `..` doesn't make sense. `Size2D { width: 123, height: 456, .. }` leaks implementation details about the internal PhantomData field and offers nothing in return.

Reply:
Given the overall design of the language, I believe that we're unlikely to have *implicit* default values, where `Size2D { width: 123, height: 456 }` can actually have many more defaulted fields. With the `const` restriction it might make sense, but it goes against the grain of surfacing "surprising" semantic behavior in the syntax.

Reply:
I think with a const restriction it's not that crazy to think we would have implicit defaults (especially for ZSTs like PhantomData), but would definitely leave it for future work.

You could imagine using unnamed fields for this, for example:
`struct Foo<'a> { _: PhantomData<&'a ()> = PhantomData }`